darc/darc.go

```diff
@@ -1180,6 +1191,103 @@ func parseIDEvmContract(in string) (Identity, error) {
 	}, nil
 }
 
+func evalThreshold(visited map[string]bool, getDarc GetDarc,
+	attrFuncs AttrInterpreters, acceptDarc bool, s string, ids []string) error {
+
+	// we have something of form 'threshold<1/2,darc:aa,ed25519:bb,...>'
+
+	s = strings.TrimPrefix(s, "threshold<")
+	s = strings.TrimSuffix(s, ">")
+
+	entries := strings.Split(s, ",")
+
+	if len(entries) < 1 {
+		return xerrors.Errorf("expected at least the threshold fraction: %s", s)
+	}
+
+	fractionStr := entries[0]
+
+	ok := regexp.MustCompile(`\d+/\d+`).MatchString(fractionStr)
+	if !ok {
+		return xerrors.Errorf("the threshold fraction is incorrect: %s", fractionStr)
+	}
+
+	fractionElem := strings.Split(fractionStr, "/")
+	numerator, err := strconv.Atoi(fractionElem[0])
+	if err != nil {
+		return xerrors.Errorf("failed to convert numerator: %v", err)
+	}
+
+	denominator, err := strconv.Atoi(fractionElem[1])
+	if err != nil {
+		return xerrors.Errorf("failed to convert denominator: %v", err)
+	}
+
+	threshold := float64(numerator) / float64(denominator)
+
+	uniqIds := map[string]struct{}{}
+	validIds := map[string]struct{}{}
```

Comment:
To avoid relocation, you can `make` it with the expected capacity of `len(entries)-1`.